client/src/common/dataProvider.ts

```diff
@@ -139,7 +142,18 @@ export const dataProvider = (pkDictionary: any, apiUrl: string): DataProvider =>
     });
   },
 
-  updateMany: (resource: string, params: any) => {},
+  updateMany: (resource: string, params: any) => {
+    const url = `${apiUrl}/${resource}/rows/${params.ids.toString()}`;
+
+    // remove the id property
+    const { id, ...editData } = params.data;
+    return axios.put(url, { fields: editData }).then(async (response) => { 
+
+      //refetch the edited items from the api
```

Comment:
@AbegaM - the React-Admin API says that `updateMany` should return an array of the ids that got updated.  If the `put` was successful, then you can just return the ids in the `params` field.
![image](https://user-images.githubusercontent.com/1108513/206132138-7ee2c1b8-eb08-46f4-b552-7ca667162481.png)

Reply:
My bad, i have seen the documentation but i haven't noticed the `Id` part, i will fix this issue

Reply:
Hopefully fixing the above will also fix the issue where the UI doesn't update until a new version comes back from the database:
![screencast](https://i.gyazo.com/36915c34d315abc99a7b55a27ee3aabf.gif)

Reply:
I have updated the `updateMany` provider, i am returning the updated `ids` but the UI is still the same, it is taking like 3 seconds to update the data

Reply:
The 3-second delay is down to `optimistic locking`:
https://marmelab.com/react-admin/doc/2.9/Actions.html#optimistic-rendering-and-undo

So, the data doesn't get sent for 3 seconds.  It appears the issue is because `react-admin` isn't updating the local state - the local state only gets updated after the backend data change.